lib/block-supports/colors.php

```diff
@@ -70,31 +77,37 @@ function gutenberg_apply_colors_support( $block_type, $block_attributes ) {
 		return array();
 	}
 
-	$has_text_colors_support       = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'text' ), true ) );
-	$has_background_colors_support = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'background' ), true ) );
-	$has_gradients_support         = _wp_array_get( $color_support, array( 'gradients' ), false );
+	$has_text_colors_support       = true === $color_support || ( is_array( $color_support ) && ( ! isset( $color_support['text'] ) || $color_support['text'] ) );
+	$has_background_colors_support = true === $color_support || ( is_array( $color_support ) && ( ! isset( $color_support['background'] ) || $color_support['background'] ) );
+	$has_gradients_support         = isset( $color_support['gradients'] ) && $color_support['gradients'];
 	$color_block_styles            = array();
 
 	// Text colors.
 	// Check support for text colors.
 	if ( $has_text_colors_support && ! wp_should_skip_block_supports_serialization( $block_type, 'color', 'text' ) ) {
 		$preset_text_color          = array_key_exists( 'textColor', $block_attributes ) ? "var:preset|color|{$block_attributes['textColor']}" : null;
-		$custom_text_color          = _wp_array_get( $block_attributes, array( 'style', 'color', 'text' ), null );
+		$custom_text_color          = isset( $block_attributes['style']['color']['text'] )
+			? _wp_array_get( $block_attributes, array( 'style', 'color', 'text' ), null )
```

Comment:
Same here. There are some more cases like that below which I won't highlight separately to avoid too many conversations. The idea is always the same, so let's discuss in one place as needed.